invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/user_records_search/components.js

```diff
@@ -37,6 +38,49 @@ import {
 
 import axios from "axios";
 
+const DeleteDraftButton = (props) => {
+  const [modalOpen, setModalOpen] = useState(false);
+
+  const handleOpen = () => setModalOpen(true);
+
+  const handleClose = () => setModalOpen(false);
+
+  const handleDelete = async () => {
+    const resp = await axios.delete(
+      `/api/records/${props.record.id}/draft`,
+      {},
+      {
+        headers: { "Content-Type": "application/json" },
+      }
+    );
+    handleClose();
+    window.location.reload();
+  };
+
+  return (
+    <>
+      <Button compact floated="right" color="red" onClick={handleOpen}>
+        <Icon name="trash alternate outline" />
+        Delete
+      </Button>
+
+      <Modal open={modalOpen} onClose={handleClose} size="tiny">
+        <Modal.Content>
+          <h3>Are you sure you want to delete this draft?</h3>
+        </Modal.Content>
+        <Modal.Actions>
```

Comment:
Question: Do we have any kind of feedback to the user on these buttons? Seems to be that we'll have a lot of buttons which sends an API request, and we should be able to show a spinner and if something went wrong. Perhaps not for this PR, but we need something similar for the "New version" button.

Reply:
We don't have at the moment but yes we need a common way to present all the API errors coming from action buttons e.g the delete, new version etc.